int32 strafe_X
int32 strafe_Y
int32 turn_X
int32 turn_Y
bool descend
bool ascend
